### PR TITLE
Adds support and test coverage for license{,s} as {string,array}

### PR DESCRIPTION
**Pull request by tmarble**
Note: disjuctinve (`OR`) of license array previously handled.

### CODE REVIEW

**Comment by geneh** [comment]
Is it related to an issue?

**Comment by tmarble** [comment]
@geneh Yes, I mentioned in the last commit message that this is for #458

File: providers/summary/clearlydefined.js [hunk 1]
```diff
@@ -244,10 +244,26 @@ class ClearlyDescribedSummarizer {
...
+    } else if (Array.isArray(manifest.license)) {
+      expression = manifest.license.join(' OR ')
+    } else if (manifest.license && (typeof manifest.license.type === 'string')) {
+      expression = manifest.license.type // handle sub-property 'type'
+    } else if (manifest.license && (Array.isArray(manifest.license.type))) {
+      expression = manifest.license.type.join(' OR ')
+    } else if (typeof manifest.licenses === 'string') {
+      expression = manifest.licenses // handle legacy NPM 'licenses' key
+    } else if (Array.isArray(manifest.licenses)) {
+      expression = manifest.licenses.join(' OR ')
```
Comment (geneh) [inline providers/summary/clearlydefined.js]: This is unclear. I've commented in the ticket to get clarification. https://github.com/clearlydefined/service/issues/458

Comment (tmarble) [reply providers/summary/clearlydefined.js]: Can you please explain what is unclear?
The idea here is the license may be expressed in the manifest under the key/property `license`,`licence.type`, or `licenses` with value datatype of array of strings (presumed to be a disjunction) or just a string.

Comment (geneh) [reply providers/summary/clearlydefined.js]: The code is clear. Please see my comment in the issue.

File: providers/summary/clearlydefined.js [hunk 2]
```diff
@@ -244,10 +244,26 @@ class ClearlyDescribedSummarizer {
         if (bugs.startsWith('http')) setIfValue(result, 'described.issueTracker', bugs)
       } else setIfValue(result, 'described.issueTracker', bugs.url || bugs.email)
     }
-    const license =
-      manifest.license &&
-      SPDX.normalize(typeof manifest.license === 'string' ? manifest.license : manifest.license.type)
-    setIfValue(result, 'licensed.declared', license)
+    // We could have singular licenses such as 'MIT' or licenses in an array ['MIT', 'BSD']
+    // Process licenses depending on whether they are strings or array of strings
+    var expression = null  // license expression (as input to SPDX.normalize)
```
Comment (geneh) [inline providers/summary/clearlydefined.js]: let or const please, no vars.

File: providers/summary/clearlydefined.js [hunk 3]
```diff
@@ -267,8 +282,18 @@ class ClearlyDescribedSummarizer {
...
     // We could have singular licenses such as 'MIT' or licenses in an array ['MIT', 'BSD']
     // Process licenses depending on whether they are strings or array of strings
-    if (Array.isArray(manifest.license)) {
-      const licenses = SPDX.normalize((manifest.license || []).join(' OR '))
+    let expression = null  // license expression (as input to SPDX.normalize)
+    if (typeof manifest.license === 'string') {
+      expression = manifest.license
+    } else if (Array.isArray(manifest.license)) {
+      expression = manifest.license.join(' OR ')
+    } else if (typeof manifest.licenses === 'string') {
```
Comment (geneh) [inline providers/summary/clearlydefined.js]: @tmarble Can PHP composer have licenses (not license) field? Is it documented anywhere? I wonder if this code will ever execute.

**Pull request merged**